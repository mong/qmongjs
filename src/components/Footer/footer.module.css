.footer {
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: center;
  background-color: #00263d;
  color: #f2f2f2;
  padding: 64px;
  align-items: center;
  width: 100%;
}

.footerTop {
  display: flex;
  flex-wrap: wrap;
  max-width: 960px;
  color: #f2f2f2;
  text-align: center;
}
footer a {
  color: #fff;
  margin-bottom: 8px;
}
.skdeContact,
.skdePrivacy {
  text-align: justify;
}
.skdeContact a {
  text-decoration: none;
  color: #7ebec7;
  font-weight: 400;
  line-height: 1.5em;
}

.skdePrivacy a {
  text-decoration: none;
  font-weight: 700;
  line-height: 1.5em;
}

.footerTop > div {
  width: 320px;
  margin-bottom: 64px;
}

.footerBottom {
  display: flex;
  flex-wrap: wrap;
  max-width: 960px;
  justify-content: flex-end;
}

.footerBottom > div {
  width: 320px;
  padding-top: 32px;
  border-top: 1px solid #27485b;
  text-align: center;
}
