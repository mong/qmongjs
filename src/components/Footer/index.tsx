import React from "react";
import { Link } from "react-router-dom";

import skdeLogo from "../../img/SKDE_hvit_lys.png";
import helseNordLogo from "../../img/hf_nord-white.svg";
import NSMLogo from "../../img/NSM_logo_hvit.png";
import helseatlasLogo from "../../img/Logo_atlas_hvit.png";

import style from "./footer.module.css";

type FooterProps = {};

const Footer: React.FC<FooterProps> = () => {
  return (
    <footer className={style.footer}>
      <div className={style.footerTop}>
        <div>
          <Link to="/">
            <img
              src={skdeLogo}
              alt="skde logo"
              width="99px"
              height="40px"
            ></img>
          </Link>
        </div>
        <div className={style.skdeContact}>
          <strong>Telefon:</strong>
          <a href="tel:77755800"> 777 55 800</a>
          <br />
          <strong>E-post:</strong>
          <a href="mailto:postmottak@helse-nord.no">
            {" "}
            postmottak@helse-nord.no
          </a>
          <br />
          <strong>Webredakt√∏r:</strong>
          <a href="mailto:webmaster@skde.no"> webmaster@skde.no</a>
          <br />
        </div>
        <div className={style.skdePrivacy}>
          <a
            title="Link til kontakt SKDE"
            href={`${window.location.origin}/kontakt`}
          >
            Kontakt
          </a>
          <br />
          <a
            title="Link til personvern"
            href={`${window.location.origin}/personvern`}
          >
            Personvern
          </a>
        </div>
      </div>
      <div className={style.footerBottom}>
        <div>
          <a title="Link til Helse Nord RHF" href="https://www.helse-nord.no">
            <img
              src={helseNordLogo}
              alt="Helse Nord logo"
              width="180px"
              height="40px"
            />
          </a>
        </div>
        <div>
          <a
            title="Link til kvalitetsregistre"
            href="https://www.kvalitetsregistre.no"
          >
            <img src={NSMLogo} alt="NSM logo" width="287px" height="32px" />
          </a>
        </div>
        <div>
          <a title="Link til helseatlas" href="https://www.helesatlas.no">
            <img
              src={helseatlasLogo}
              alt="Helseatlas logo"
              width="146px"
              height="40px"
            />
          </a>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
