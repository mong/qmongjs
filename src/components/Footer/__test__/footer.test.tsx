import { BrowserRouter, Route } from "react-router-dom";
import { QueryParamProvider } from "use-query-params";
import { render } from "@testing-library/react";

import Footer from "../";

it("Footer renders", async () => {
  const { container } = render(
    <BrowserRouter>
      <QueryParamProvider ReactRouterRoute={Route}>
        <Footer />
      </QueryParamProvider>
    </BrowserRouter>
  );

  expect(container).toMatchSnapshot();
});
