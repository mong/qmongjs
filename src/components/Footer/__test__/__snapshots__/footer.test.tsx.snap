// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footerTop"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="skde logo"
            height="40px"
            src="SKDE_hvit_lys.png"
            width="99px"
          />
        </a>
      </div>
      <div
        class="skdeContact"
      >
        <strong>
          Telefon:
        </strong>
        <a
          href="tel:77755800"
        >
           777 55 800
        </a>
        <br />
        <strong>
          E-post:
        </strong>
        <a
          href="mailto:postmottak@helse-nord.no"
        >
           
          postmottak@helse-nord.no
        </a>
        <br />
        <strong>
          Webredakt√∏r:
        </strong>
        <a
          href="mailto:webmaster@skde.no"
        >
           webmaster@skde.no
        </a>
        <br />
      </div>
      <div
        class="skdePrivacy"
      >
        <a
          href="https://helse-nord.no/skde/kontakt-skde"
          title="Link til kontakt SKDE"
        >
          Kontakt
        </a>
        <br />
        <a
          href="https://helse-nord.no/om-nettstedet/personvern"
          title="Link til personvern"
        >
          Personvern
        </a>
      </div>
    </div>
    <div
      class="footerBottom"
    >
      <div>
        <a
          href="https://www.helse-nord.no"
          title="Link til Helse Nord RHF"
        >
          <img
            alt="Helse Nord logo"
            height="40px"
            src="hf_nord-white.svg"
            width="180px"
          />
        </a>
      </div>
      <div>
        <a
          href="https://www.kvalitetsregistre.no"
          title="Link til kvalitetsregistre"
        >
          <img
            alt="NSM logo"
            height="40px"
            src="NSM_logo_hvit.png"
            width="359px"
          />
        </a>
      </div>
      <div>
        <a
          href="https://www.helesatlas.no"
          title="Link til helseatlas"
        >
          <img
            alt="Helseatlas logo"
            height="40px"
            src="Logo_atlas_hvit.png"
            width="146px"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
